job_application/models.py

```diff
@@ -4,9 +4,25 @@
 
 
 class Application(models.Model):
-    job = models.ForeignKey(Job, on_delete=models.CASCADE, related_name='application_job')
-    student = models.ForeignKey(Student, on_delete=models.CASCADE, related_name='application_student')
+    job = models.ForeignKey(Job, on_delete=models.CASCADE,
+                            related_name='application_job')
+    student = models.ForeignKey(
+        Student, on_delete=models.CASCADE, related_name='application_student')
     date_applied = models.DateField()
 
+    @classmethod
+    def get_applications_by_job(cls, job):
+        """
+        Allows recuriters to search for all of the applications for some job.
+        """
+        return cls.objects.filter(job_id=job.id)
+
+    @classmethod
+    def get_applications_by_student(cls, student):
+        """
+        Allows students to search for all of their job applications.
+        """
+        return cls.objects.filter(student_id=student.user.id)
```

Comment:
Are you sure that's working? I'm not sure if Student has an id field because we overrode the PK in Student.
Anyway even if it does, you're comparing the id of a student to the id of the auth user that's associated with the student, these are not the same ids.
I think changing it to 
`return cls.objects.filter(student_user_id = student.user.id)`
is better.

Reply:
Yes it's working we talked about it in The slack, I send you an image of your method and that's send an error.

Reply:
My bad, the correct syntax is
`return cls.objects.filter(student__user_id=student.user.id)`
Notice the double underscore between student and user.
It's working like that I just tested it.